sunpy/net/dataretriever/sources/goes.py

```diff
@@ -141,16 +145,27 @@ def _get_metalist(self, matchdict):
         # The data before the re-processed GOES 8-15 data.
         if (matchdict["End Time"] < "2001-03-01") or (matchdict["End Time"] >= "2001-03-01" and matchdict["Provider"] == ["sdac"]):
             metalist += self._get_metalist_fn(matchdict, self.baseurl_old, self.pattern_old)
-        # New data from NOAA.
+        # New data from NOAA. It searches for both the high cadence and 1 minute average data.
         else:
             if matchdict["End Time"] >= "2017-02-07":
                 for sat in [16, 17]:
-                    metalist += self._get_metalist_fn(matchdict,
-                                                      self.baseurl_r.format(SatelliteNumber=sat), self.pattern_r)
+                    for res in matchdict["Resolution"]:
+                        metalist += self._get_metalist_fn(matchdict,
+                                                          self.baseurl_r.format(SatelliteNumber=sat, Resolution=res), self.pattern_r)
             if matchdict["End Time"] <= "2020-03-04":
                 for sat in [8, 9, 10, 11, 12, 13, 14, 15]:
-                    metalist += self._get_metalist_fn(matchdict,
-                                                      self.baseurl_new.format(SatelliteNumber=sat), self.pattern_new)
+                    # The 1 minute average data is at a different base URL to that of the high cadence data which is why things are done this way.
+                    for res in matchdict["Resolution"]:
+                        if res == "avg1m":
+                            filename_res = "xrsf"
+                            resolution = "avg1m"
+                            metalist += self._get_metalist_fn(matchdict,
+                                                              self.baseurl_new.format(SatelliteNumber=sat, filename_res=filename_res, resolution=resolution), self.pattern_new)
+                        else:
+                            filename_res = "gxrs"
+                            resolution = "irrad"
+                            metalist += self._get_metalist_fn(matchdict,
+                                                              self.baseurl_new.format(SatelliteNumber=sat, filename_res=filename_res, resolution=resolution), self.pattern_new)
```

Comment:
Similar to the above comment, I think it is worth being a bit safer here and erroring if 'res' is not "avg1m" or "irrad"